packages/contracts/contracts/L1/interfaces/IAssetManager.sol

```diff
@@ -294,151 +249,166 @@ interface IAssetManager {
+     * @return The total number of KGHs held by the vault.
      */
-    function previewDelegate(address validator, uint128 assets) external view returns (uint128);
+    function totalKghNum(address validator) external view returns (uint128);
 
     /**
-     * @notice Allows an on-chain or off-chain user to simulate the effects of their KRO undelegation
-     *         at the current block.
+     * @notice Returns the total amount of KRO in KGHs held by the vault.
      *
-     * @param validator The address of the validator.
-     * @param shares    The amount of shares to undelegate.
+     * @param validator Address of the validator.
      *
-     * @return The amount of assets that the Vault would exchange for the amount of shares provided.
+     * @return The total amount of KRO in KGHs held by the vault.
      */
-    function previewUndelegate(address validator, uint128 shares) external view returns (uint128);
+    function totalKroInKgh(address validator) external view returns (uint128);
 
     /**
-     * @notice Allows an on-chain or off-chain user to simulate the effects of their KGH delegation
-     *         at the current block given current on-chain conditions.
+     * @notice Returns the amount of KRO shares that the KRO delegator has.
      *
-     * @param validator The address of the validator.
+     * @param validator Address of the validator.
+     * @param delegator Address of the delegator.
      *
-     * @return The amount of shares that the Vault would exchange for the amount of assets provided.
+     * @return The amount of KRO shares that the KRO delegator has.
      */
-    function previewKghDelegate(address validator) external view returns (uint128);
+    function getKroTotalShareBalance(
+        address validator,
+        address delegator
+    ) external view returns (uint128);
 
     /**
-     * @notice Allows an on-chain or off-chain user to simulate the effects of their KGH undelegation
-     *         at the current block, given current on-chain conditions.
+     * @notice Returns when the KRO delegators can undelegate KRO. The delegators can undelegate
+     *         after UNDELEGATION_DELAY_SECONDS elapsed from lastDelegatedAt.
      *
-     * @param validator The address of the validator.
-     * @param delegator The address of the delegator.
-     * @param tokenId   The tokenId of KGH to undelegate.
+     * @param validator Address of the validator.
+     * @param delegator Address of the KRO delegator.
      *
-     * @return The amount of assets that the Vault would exchange for the KGH of given tokenId.
+     * @return When the KRO delegators can undelegate KRO.
      */
-    function previewKghUndelegate(
+    function canUndelegateKroAt(
         address validator,
-        address delegator,
-        uint256 tokenId
-    ) external view returns (uint128);
+        address delegator
+    ) external view returns (uint256);
```

Comment:
Do you think it's better to be uint128?

Reply:
Unify the types. `uint128` is used above.

Reply:
I was just wondering if it will be used in other parts like in ValidatorManager. Then `uint128` will be better, but if it is only used for retrieving information on frontend, `uint256` fits better imo.

Reply:
I changed the return type for `can~At` functions to `uint128`. I thought these view functions can be used in `AssetManager` as `canFinalizeCommissionChangeAt` used in `ValidatorManager`.

Reply:
Btw then shouldn't it be public, not external function?

Reply:
Yep but since it's interface, the function cannot be public. We need to change it to public in implementation contract.